csharp/extractor/Semmle.Extraction/FilePattern.cs

```diff
@@ -76,8 +76,20 @@ private static StringBuilder BuildRegex(string pattern)
                             throw new InvalidFilePatternException(pattern, "'**' preceeded by non-`/` character.");
                         if (HasCharAt(i + 2, c => c != '/'))
                             throw new InvalidFilePatternException(pattern, "'**' succeeded by non-`/` character");
-                        sb.Append(".*");
-                        i += 2;
+
+                        if (i + 2 < pattern.Length)
+                        {
+                            // Processing .../**/...
+                            sb.Append("(.*/|)");
```

Comment:
I had took look at this a while for parsing it: It is an alternation construct that matches any sequence of characters that ends with `/` OR "nothing".

Reply:
Yes, that's the idea.